src/main/scala/com/campudus/tableaux/HelperFunctions.scala

```diff
@@ -1,45 +1,107 @@
 package com.campudus.tableaux
 
 import org.vertx.scala.core.json.{ JsonObject, JsonArray }
-import com.campudus.tableaux.database.Mapper
+import com.campudus.tableaux.ArgumentChecker._
+import com.campudus.tableaux.database._
 
 object HelperFunctions {
 
-  def jsonToSeqOfRowsWithColumnIdAndValue(json: JsonObject): Seq[Seq[(Long, _)]] = {
+  private def asCastedList[A](array: JsonArray): ArgumentCheck[Seq[A]] = {
     import scala.collection.JavaConverters._
+    sequence(array.asScala.toList.map(notNull(_, "some array value").flatMap(castElement[A])))

```

Comment:
We should use `toStream` instead of `toList`, if possible. Otherwise it will go through the list twice - once for `map` and once for `flatMap`.